src/posit/connect/client.py

```diff
@@ -37,13 +46,20 @@ def __init__(
 
         # Place to cache the server settings
         self.server_settings = None
+        self._oauth = None
 
     @property
     def connect_version(self):
         if self.server_settings is None:
             self.server_settings = self.get("server_settings").json()
         return self.server_settings["version"]
 
+    @property
+    def oauth(self) -> OAuthIntegration:
+        if self._oauth is None:
+            self._oauth = OAuthIntegration(config=self.config, session=self.session)
+        return self._oauth
```

Comment:
AFAICT there's nothing in `OAuthIntegration` that needs to be cached, so we just need to be able to access it here, as we do (now) for `Users` below.

```suggestion
        return OAuthIntegration(config=self.config, session=self.session)
```